transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -203,6 +203,7 @@ def general_grouped_gemm(
             for o in out
         ]  # this should differ with respect to single output
 
+    original_scale_inverses_list = [swizzle_inputs(A[i], B[i], layout) for i in range(num_gemms)]
```

Comment:
A weird thing here is that when I attempted to swizzle the scales in C++, i.e., inserting two lines
```cpp
    swizzle_scaling_factors(te_A, transa);
    swizzle_scaling_factors(te_B, !transb);
```
after
https://github.com/NVIDIA/TransformerEngine/blob/97344d669670234de6517053b3eebefb05e3a40b/transformer_engine/pytorch/csrc/extensions/gemm.cpp#L325-L327
it will give incorrect results. I'm quite confusing.

Reply:
@ptrendx This looks like the same issue I encountered as well

Reply:
From a quick look into it - that's because swizzle_scaling_factors is not doing the right thing - it creates a pyTorch tensor for the new scaling factor but then lets it go out of scope releasing the memory.
```python
  auto swizzled_scale_inv = at::empty(scale_inv_shape_int, options);
```
This tensor should be returned by that swizzle function to make sure the memory is still there when the caller uses it.

Reply:
Thanks. That makes sense. I think we can put the fix in another PR.